### ISSUE TITLE
When clicking on video, only sound is played

### ISSUE THREAD

**Issue by kuzmos** [LibrePhotos/librephotos#379]
Video thumbnail is generated and displayed correctly when scrolling through photos and videos on the "Photos" page, but when I click on the video, only a gray transparent overlay is displayed with sound, but no picture. When I right-click on the video and select full-screen, it is played correctly. Is it maybe some css issue?

**Comment by kuzmos** [comment]
`lsb_release -a`
`LSB Version:	core-11.1.0ubuntu2-noarch:printing-11.1.0ubuntu2-noarch:security-11.1.0ubuntu2-noarch`
`Distributor ID:	Ubuntu`
`Description:	Ubuntu 20.04.3 LTS`
`Release:	20.04`
`Codename:	focal`

Firefox 93.0 (64-bit)

**Comment by Amarel-LTD** [comment]
I have the same issue as well.

**Comment by kuzmos** [comment]
Same issue is happening while browsing on Firefox mobile on Android. A gray transparent rectangle overlay with sound and progress bar, but no video.

**Comment by derneuere** [comment]
It works with my video files for firefox and chrome. What kind of video file is it?

**Comment by ErgoProxy1337** [comment]
same here. for me that are .mov files. While holding with the mouse over the video, I can see the video play but when clicking on it, I get only audio

**Comment by derneuere** [comment]
Yes, the previews get reencoded to be in the right format all of the time.
Do you know which codec this file uses? You can find it with VLC: https://superuser.com/questions/348159/vlc-how-do-i-see-video-files-information-resolution-encoding-information-e/1226072

**Comment by ErgoProxy1337** [comment]
![grafik](https://user-images.githubusercontent.com/38110913/139252709-4bc1e0b9-732f-4f73-ac07-9bc6022907f2.png)
Mhm okey... h265 is hard...

**Comment by derneuere** [comment]
Do you have a h264 encoded video? Does that work?

**Comment by ErgoProxy1337** [comment]
No, sry. Those are photos taken from my iphone. Thats the default format. 
Btw. I dont think it is exactly the same problem as the thread creator has because I dont have on this video any full screen button like on the normal ones. So I have an different problem related maybe to the H265 codec.

For normal videos like mp4, it works on normal mode and also on full screen.

**Comment by kuzmos** [comment]
My videos are taken by Samsung s10+ in h265 codec, either fullHD or 4K.

**Comment by derneuere** [comment]
Alright! I added an experimental setting in the latest version. It's called "Always transcode videos". This will activate transcoding. 

This is not yet ready for production use, but it would be useful for me, if you could test video playback with this setting. Please remember to deactivated again after testing :)

**Comment by kuzmos** [comment]
Updated, enabled the feature, let's see what the night re-scan gives us in the morning...

**Comment by kuzmos** [comment]
> Updated, enabled the feature, let's see what the night re-scan gives us in the morning...

My bad, realized don't need to wait for the re-scan. When I click on a video now, it starts playing - as expected, however the sound quality is superbad, and the video quality is not stellar. Also, which is worse, I can't access the full-quality version, or I don't know how - when I click on info -> file path, or right-click -> full-screen, it is still the same transcoded-quality video. ffmpeg process appears on the server of course. But definitely a step in the right direction.

**Comment by marcobehler** [comment]
Having the same issue as well, browser doesn't matter (edge, brave, firefox..). How/where can I enable video transcoding?

**Comment by marcobehler** [comment]
Interestingly, I can play older videos, i.e. only "newer" videos are broken - maybe the mobile phone/cam had an update. In any case, here's the mediainfo for the newish, not-working .mp4

> `Format                                   : MPEG-4
> Format profile                           : Base Media
> Codec ID                                 : isom (isom/iso2/mp41)
> File size                                : 3.05 MiB
> Duration                                 : 1 s 190 ms
> Overall bit rate                         : 21.5 Mb/s
> Encoded date                             : UTC 2021-12-17 16:02:10
> Tagged date                              : UTC 2021-12-17 16:02:10
> xyz                                      : +48.0989+11.5435/
> 
> Video
> ID                                       : 3
> Format                                   : HEVC
> Format/Info                              : High Efficiency Video Coding
> Format profile                           : Main@L5.1@Main
> Codec ID                                 : hvc1
> Codec ID/Info                            : High Efficiency Video Coding
> Duration                                 : 1 s 190 ms
> Bit rate                                 : 21.3 Mb/s
> Width                                    : 1 920 pixels
> Height                                   : 1 080 pixels
> Display aspect ratio                     : 16:9
> Rotation                                 : 270°
> Frame rate mode                          : Variable
> Frame rate                               : 30.253 FPS
> Minimum frame rate                       : 29.970 FPS
> Maximum frame rate                       : 42.857 FPS
> Standard                                 : NTSC
> Color space                              : YUV
> Chroma subsampling                       : 4:2:0
> Bit depth                                : 8 bits
> Bits/(Pixel*Frame)                       : 0.339
> Stream size                              : 3.02 MiB (99%)
> Title                                    : VideoHandle
> Language                                 : English
> Encoded date                             : UTC 2021-12-17 16:02:10
> Tagged date                              : UTC 2021-12-17 16:02:10
> Color range                              : Full
> Color primaries                          : BT.601 PAL
> Transfer characteristics                 : BT.709
> transfer_characteristics_Original        : BT.601
> Matrix coefficients                      : BT.601
> matrix_coefficients_Original             : BT.470 System B/G
> Codec configuration box                  : hvcC
> 
> Audio
> ID                                       : 2
> Format                                   : AAC LC
> Format/Info                              : Advanced Audio Codec Low Complexity
> Codec ID                                 : mp4a-40-2
> Duration                                 : 1 s 166 ms
> Bit rate mode                            : Constant
> Bit rate                                 : 192 kb/s
> Channel(s)                               : 2 channels
> Channel layout                           : L R
> Sampling rate                            : 48.0 kHz
> Frame rate                               : 46.875 FPS (1024 SPF)
> Compression mode                         : Lossy
> Stream size                              : 26.7 KiB (1%)
> Title                                    : SoundHandle
> Language                                 : English
> Encoded date                             : UTC 2021-12-17 16:02:10
> Tagged date                              : UTC 2021-12-17 16:02:10
> 
> Other
> Type                                     : meta
> Duration                                 : 1 s 190 ms
> Bit rate mode                            : Variable`

And here's the mediainfo for a mp4 that's working:

> `Format                                   : MPEG-4
> Format profile                           : Base Media / Version 2
> Codec ID                                 : mp42 (isom/mp42)
> File size                                : 140 MiB
> Duration                                 : 35 s 456 ms
> Overall bit rate                         : 33.2 Mb/s
> Encoded date                             : UTC 2020-12-20 11:37:59
> Tagged date                              : UTC 2020-12-20 11:37:59
> xyz                                      : +48.1263+11.5637/
> 
> Video
> ID                                       : 3
> Format                                   : AVC
> Format/Info                              : Advanced Video Codec
> Format profile                           : High@L5.1
> Format settings                          : CABAC / 1 Ref Frames
> Format settings, CABAC                   : Yes
> Format settings, Reference frames        : 1 frame
> Format settings, GOP                     : M=1, N=60
> Codec ID                                 : avc1
> Codec ID/Info                            : Advanced Video Coding
> Duration                                 : 35 s 440 ms
> Bit rate                                 : 33.0 Mb/s
> Width                                    : 1 920 pixels
> Height                                   : 1 080 pixels
> Display aspect ratio                     : 16:9
> Frame rate mode                          : Variable
> Frame rate                               : 60.000 FPS
> Minimum frame rate                       : 55.249 FPS
> Maximum frame rate                       : 63.425 FPS
> Standard                                 : NTSC
> Color space                              : YUV
> Chroma subsampling                       : 4:2:0
> Bit depth                                : 8 bits
> Scan type                                : Progressive
> Bits/(Pixel*Frame)                       : 0.265
> Stream size                              : 140 MiB (99%)
> Title                                    : VideoHandle
> Language                                 : English
> Encoded date                             : UTC 2020-12-20 11:37:59
> Tagged date                              : UTC 2020-12-20 11:37:59
> Color range                              : Full
> Color primaries                          : BT.601 PAL
> Transfer characteristics                 : BT.709
> transfer_characteristics_Original        : BT.601
> Matrix coefficients                      : BT.601
> Codec configuration box                  : avcC
> 
> Audio
> ID                                       : 2
> Format                                   : AAC LC
> Format/Info                              : Advanced Audio Codec Low Complexity
> Codec ID                                 : mp4a-40-2
> Duration                                 : 35 s 425 ms
> Bit rate mode                            : Constant
> Bit rate                                 : 192 kb/s
> Channel(s)                               : 2 channels
> Channel layout                           : L R
> Sampling rate                            : 48.0 kHz
> Frame rate                               : 46.875 FPS (1024 SPF)
> Compression mode                         : Lossy
> Stream size                              : 830 KiB (1%)
> Title                                    : SoundHandle
> Language                                 : English
> Encoded date                             : UTC 2020-12-20 11:37:59
> Tagged date                              : UTC 2020-12-20 11:37:59
> 
> Other
> Type                                     : meta
> Duration                                 : 35 s 456 ms
> Bit rate mode                            : Variable`

**Comment by Janhouse** [comment]
This can be closed now. Latest version allows to enable transcoding of videos.

**Comment by derneuere** [comment]
Related to #365

**Comment by derneuere** [comment]
Resolved by the video transcoding feature: `transcode_videos` is now a persisted user setting, which fixes the codec cases (HEVC/H.265) behind this report.
